ocs_ci/utility/azure_utils.py

```diff
@@ -1,51 +1,180 @@
+# -*- coding: utf8 -*-
+"""
+Module for interactions with OCP/OCS Cluster on Azure platform level.
+"""
+
+import json
 import logging
+import os
 
 from azure.common.credentials import ServicePrincipalCredentials
 from azure.mgmt.compute import ComputeManagementClient
-from ocs_ci.utility.utils import TimeoutSampler
+from azure.mgmt.resource import ResourceManagementClient
+
+
+from ocs_ci.framework import config
 from ocs_ci.ocs.exceptions import TimeoutExpiredError
+from ocs_ci.utility.utils import TimeoutSampler
 
 logger = logging.getLogger(name=__file__)
 
 
+# default location of files with necessary azure cluster details
+SERVICE_PRINCIPAL_FILEPATH = os.path.expanduser("~/.azure/osServicePrincipal.json")
+TERRRAFORM_FILENAME = "terraform.azure.auto.tfvars.json"
+
+
+def load_cluster_resource_group(
+    cluster_path,
+    terraform_filename=TERRRAFORM_FILENAME
+):
+    """
+    Read terraform tfvars.json file created by ``openshift-installer`` in a
+    cluster dir to get azure ``resource group`` of an OCP cluster. All Azure
+    resources of the cluster are placed in this group.
+
+    Args:
+        cluster_path (str): full file path of the openshift cluster directory
+        terraform_filename (str): name of azure terraform vars file, this is
+            optional and you need to specify this only if you want to override
+            the default
+
+    Returns:
+        string with resource group name
```

Comment:
Should be:

```python
    Returns:
        str: The resource group name

"""
```

Reply:
please apply this in the other functions as well

Reply:
ack

Reply:
I decided to leave it as it is to save my time. On one hand [we note in the code style guide that we should follow google style python docstrings](https://ocs-ci.readthedocs.io/en/latest/docs/coding_guidelines.html), referencing [examples which add these newlines](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html), but besides the exaple, there is no explicit requirement for this particular style there (I discussed this some time ago with @petr-balogh ), neither we check it during tox run, and [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html) itself doesn't do that.

Reply:
It may not be enforced or required by google's docstring format but it is definitely the convention that we have chosen to adopt in our code.

Another convention we have chosen to follow is for docstrings with additional sections (Args, Raises, Returns, etc.) is adding an empty line at the end of the docstring.

Our [coding guidelines](https://ocs-ci.readthedocs.io/en/latest/docs/coding_guidelines.html) has an example docstring which does both of these, but we should probably add a couple points into the docstring section to be more explicit.